client/src/components/pages.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropType from 'prop-types';
+
+class Pages extends React.Component {
+  next() {
+    if (this.props.currentPage < this.props.numberOfPages - 1) {
+      return <button onClick={this.props.handleNextClick}>Next</button>;
+    }
+    return null;
+  }
+
+  prev() {
+    if (this.props.currentPage !== 0) {
+      return <button onClick={this.props.handlePrevClick}>Prev</button>;
+    }
+    return null;
+  }
+
+  loop(start, end) {
+    const pages = [];
+    if (end <= this.props.numberOfPages) {
+      for (let i = start; i <= end; i++) {
+        pages.push(<button value={i} key={i} onClick={this.props.handlePageClick}>{i}</button>);
+      }
+    } else {
+      for (let i = start; i < this.props.numberOfPages; i++) {
+        pages.push(<button value={i} key={i} onClick={this.props.handlePageClick}>{i}</button>);
+      }
+    }
+    return pages;
+  }
+
+  renderEndingDots() {
+    if (this.props.currentPage < this.props.numberOfPages - 4) {
+      return <div>...</div>;
+    }
+    return null;
+  }
+  renderBeginningDots() {
+    if (this.props.currentPage > 3) {
+      return <div>...</div>;
+    }
+    return null;
+  }
+
+  renderPages(currentPage) {
+    const pages = this.props.numberOfPages;
+    if (currentPage < 2) {
```

Comment:
This was a bit difficult to follow the logic here without a visual or explanation, but I take it you tested it out and it works for all number of reviews and pages.

Reply:
Yeah this wouldn't make a lot of sense without seeing it. Thanks for the review!